rules/DeadCode/Rector/Cast/RecastingRemovalRector.php

```diff
@@ -103,6 +105,13 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
+        // substr can return false on php 7.x
+        if ($node->expr instanceof FuncCall
+            && $this->isName($node->expr, 'substr')
+            && ! $node->expr->isFirstClassCallable()) {
+            $nodeType = new UnionType([new StringType(), new ConstantBooleanType(false)]);
+        }
```

Comment:
This should be checked by version in Rector. If on PHP 8.0+, this should be skipped
+ downgrade rule added to make it safe

Reply:
There is no need php version check, as phpstan already check it when run on php 7.x, just need add skip config in `rector.php` for now:

```php
        // on php 7.x, substr() result can return false, so force (string) is needed
        RecastingRemovalRector::class => [
            __DIR__ . '/rules/CodingStyle/ClassNameImport/ClassNameImportSkipVoter/ClassLikeNameClassNameImportSkipVoter.php',
        ],
```